server/errors/errors.go

```diff
@@ -0,0 +1,174 @@
+package errors
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"github.com/fleetdm/fleet/v4/server/datastore/redis"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	kitlog "github.com/go-kit/kit/log"
+	redigo "github.com/gomodule/redigo/redis"
+	"github.com/mna/redisc"
+	"github.com/rotisserie/eris"
+)
+
+var errCh chan error
+var running int32
+
+type errorHandler struct {
+	pool   fleet.RedisPool
+	logger kitlog.Logger
+}
+
+type ErrorFlusher interface {
+	Flush() ([]string, error)
+}
+
+func NewHandler(ctx context.Context, pool fleet.RedisPool, logger kitlog.Logger) ErrorFlusher {
+	errCh = make(chan error)
```

Comment:
It should be buffered so that creating a new error does not block for sync with the handleErrors goroutine.

```suggestion
	errCh = make(chan error, 1)
```

Reply:
But even if it's buffered, fleet can still produce more than 1 error and block in the same way. That's why New() has a timeout.

Reply:
Yeah for sure, I think both should be used - the buffered of one to ensure it can go in without having to sync, and the timeout to protect against a sudden overflow of errors to handle.